### ISSUE TITLE
HDF5: attributes are not supported

### ISSUE THREAD

**Issue by FRidh** [blaze/blaze#1606]
I just found out that [attributes](http://docs.h5py.org/en/latest/high/attr.html) of a dataset are not copied by `odo`. Looking at the dataset with `Data` I noticed attributes don't show up either.
As described in the link, attributes are pretty important for users of HDF5.

**Comment by ekaschalk** [comment]
Agreed, can't use hdf5 file format idiomatically without attributes. Need to create annoying meta groups which obfuscates the structure.

**Comment by llllllllll** [comment]
How would you want the attributes represented? Right now blaze and odo unify different backends through the abstraction of a table, so we don't preserve metadata across queries or conversions. How would you want to query the attributes?

One option would be to use `odo.resource` to get the hdf object and then read the attributes from that. You could create the `bz.data` wrapper around this object directly so that you have access to both the attrs and the expressions.

**Comment by FRidh** [comment]
> How would you want the attributes represented? Right now blaze and odo unify different backends through the abstraction of a table, so we don't preserve metadata across queries or conversions. How would you want to query the attributes?

I think there should be a reserved attribute that can hold a dictionary with metadata. I don't know whether that's doable with `blaze` though since it deals with tables. 

Relevant links:
- pandas: https://github.com/pandas-dev/pandas/issues/2485
- xarray: http://xarray.pydata.org/en/stable/faq.html#approach-to-metadata
